mesonbuild/options.py

```diff
@@ -106,12 +106,13 @@ class OptionKey:
     internally easier to reason about and produce.
     """
 
-    __slots__ = ['name', 'subproject', 'machine', '_hash']
+    __slots__ = ['name', 'subproject', 'machine', '_hash', '_as_tuple']
```

Comment:
I'm really not sure about this one. Tuple representation is only used for comparison, and I doubt we really need to sort OptionKeys. Especially since we now use the hash for equality comparison.